app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.ts

```diff
@@ -0,0 +1,220 @@
+};
+
+const defaultState: MetaMetricsDataDeletionState = {
+  metaMetricsDataDeletionId: '',
+  metaMetricsDataDeletionDate: 0,
+};
+
+// Metadata for the controller state
+const metadata = {
+  metaMetricsDataDeletionId: {
+    persist: true,
+    anonymous: true,
+  },
+  metaMetricsDataDeletionDate: {
+    persist: true,
+    anonymous: true,
+  },
+  metaMetricsDataDeletionStatus: {
+    persist: true,
+    anonymous: true,
+  },
+  hasMetaMetricsDataRecorded: {
+    persist: true,
+    anonymous: true,
+  },
+};
+
+// Describes the action creating the delete regulation task
+export type CreateMetaMetricsDataDeletionTaskAction = {
+  type: `${typeof controllerName}:createMetaMetricsDataDeletionTask`;
+  handler: MetaMetricsDataDeletionController['createMetaMetricsDataDeletionTask'];
+};
+
+// Describes the action to check the existing regulation status
+export type UpdateDataDeletionTaskStatusAction = {
+  type: `${typeof controllerName}:updateDataDeletionTaskStatus`;
+  handler: MetaMetricsDataDeletionController['updateDataDeletionTaskStatus'];
+};
+
+// Describes the action to records whether data was collected after last deletion
+export type SetHasMetaMetricsDataRecordedAction = {
+  type: `${typeof controllerName}:setHasMetaMetricsDataRecorded`;
+  handler: MetaMetricsDataDeletionController['setHasMetaMetricsDataRecorded'];
+};
+
+// Union of all possible actions for the messenger
+export type MetaMetricsDataDeletionControllerMessengerActions =
+  | CreateMetaMetricsDataDeletionTaskAction
+  | UpdateDataDeletionTaskStatusAction
+  | SetHasMetaMetricsDataRecordedAction;
+
+// Type for the messenger of MetaMetricsDataDeletionController
+export type MetaMetricsDataDeletionControllerMessenger =
+  RestrictedControllerMessenger<
+    typeof controllerName,
+    MetaMetricsDataDeletionControllerMessengerActions,
+    never,
+    never,
+    never
+  >;
+
+/**
+ * Controller responsible for maintaining
+ * state related to Metametrics data deletion
+ */
+export class MetaMetricsDataDeletionController extends BaseController<
+  typeof controllerName,
+  MetaMetricsDataDeletionState,
+  MetaMetricsDataDeletionControllerMessenger
+> {
+  #dataDeletionService: PublicInterface<DataDeletionService>;
+
+  #getMetaMetricsId: () => string | null;
+
+  /**
+   * Creates a MetaMetricsDataDeletionController instance.
+   *
+   * @param args - The arguments to this function.
+   * @param args.dataDeletionService - The service used for deleting data.
+   * @param args.messenger - Messenger used to communicate with BaseV2 controller.
+   * @param args.state - Initial state to set on this controller.
+   * @param args.getMetaMetricsId - A function that returns the current MetaMetrics ID.
+   */
+  constructor({
+    dataDeletionService,
+    messenger,
+    state,
+    getMetaMetricsId,
+  }: {
+    dataDeletionService: PublicInterface<DataDeletionService>;
+    messenger: MetaMetricsDataDeletionControllerMessenger;
+    state?: MetaMetricsDataDeletionState;
```

Comment:
```suggestion
    state?: Partial<MetaMetricsDataDeletionState>;
```

It should be safe to supply an incomplete set of state properties, since we're populating the state object with `defaultState` first.